lib/WWW/Mechanize.pm

```diff
@@ -2129,6 +2135,12 @@ sub submit_form {
         $self->{current_form} = $matched[0];
     }
 
+    if (defined($args{strict_forms})) {
```

Comment:
if (defined($args{strict_forms})) {
        # Strict argument has been passed, set the flag as appropriate
        # this must be done prior to attempting to set the fields
        $self->current_form->strict($args{strict_forms});
}

If the value is defined it will pass through whatever you have specified, so if you pass strict_forms => 0 then it will set strict to 0 which HTML::Form will interpret as false and not apply strict.